lib/pangea/controllers/message_analytics_controller.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:fluffychat/pangea/controllers/get_analytics_controller.dart';
+import 'package:fluffychat/pangea/enum/activity_type_enum.dart';
+import 'package:fluffychat/pangea/matrix_event_wrappers/pangea_message_event.dart';
+import 'package:fluffychat/pangea/models/analytics/constructs_model.dart';
+import 'package:fluffychat/pangea/models/practice_activities.dart/message_activity_request.dart';
+
+/// Picks which tokens to do activities on and what types of activities to do
+/// Caches result so that we don't have to recompute it
+/// Most importantly, we can't do this in the state of a message widget because the state is disposed of and recreated
+/// If we decided that the first token should have a hidden word listening, we need to remember that
+/// Otherwise, the user might leave the chat, return, and see a different word hidden
+class MessageAnalyticsEntry {
+  final DateTime createdAt = DateTime.now();
```

Comment:
Where does a comment here go?

Reply:
You'd see it when you hover over the class name

Reply:
That's pretty cool. Maybe we should be using this over some types of code comments.